tests/test_search_optimizer.c

```
@@ -0,0 +1,164 @@
+    CLEED_TEST_ASSERT(cfg.seed == 789);
+
+    CLEED_TEST_ASSERT(test_unset_env_value("CSEARCH_MAX_EVALS") == 0);
+    CLEED_TEST_ASSERT(test_unset_env_value("CSEARCH_MAX_ITERS") == 0);
+    CLEED_TEST_ASSERT(test_unset_env_value("CSEARCH_SEED") == 0);
+
+    return 0;
+}
+
+static int test_config_apply(void)
+{
+    const int orig_amoeba = sr_amoeba_eval_limit;
```

Clang (14.0.6)

[warning] variable 'orig_amoeba' is not initialized

(cppcoreguidelines-init-variables)